internal/controller/apiextensions/composite/environment/selector.go

```diff
@@ -106,28 +112,33 @@ func (s *APIEnvironmentSelector) buildEnvironmentConfigRefFromRef(ref *v1.Enviro
 	}
 }
 
-func (s *APIEnvironmentSelector) buildEnvironmentConfigRefFromSelector(ctx context.Context, cr resource.Composite, selector *v1.EnvironmentSourceSelector) (corev1.ObjectReference, error) {
+func (s *APIEnvironmentSelector) buildEnvironmentConfigRefFromSelector(ctx context.Context, cr resource.Composite, selector *v1.EnvironmentSourceSelector) ([]corev1.ObjectReference, error) {
 	matchLabels := make(client.MatchingLabels, len(selector.MatchLabels))
 	for i, m := range selector.MatchLabels {
 		val, err := ResolveLabelValue(m, cr)
 		if err != nil {
-			return corev1.ObjectReference{}, errors.Wrapf(err, errFmtResolveLabelValue, i)
+			return []corev1.ObjectReference{}, errors.Wrapf(err, errFmtResolveLabelValue, i)
 		}
 		matchLabels[m.Key] = val
 	}
 	res := &v1alpha1.EnvironmentConfigList{}
 	if err := s.kube.List(ctx, res, matchLabels); err != nil {
-		return corev1.ObjectReference{}, errors.Wrap(err, errListEnvironmentConfigs)
+		return []corev1.ObjectReference{}, errors.Wrap(err, errListEnvironmentConfigs)
 	}
 	if len(res.Items) == 0 {
-		return corev1.ObjectReference{}, errors.New(errListEnvironmentConfigsNoResult)
+		return []corev1.ObjectReference{}, errors.New(errListEnvironmentConfigsNoResult)
```

Comment:
I know it's technically not part of this change, but it seems like it should be valid for a Selector to return no results, just as it is valid for a CompositionSelector to return no results and fallback to the defaultComposition.  In this case the environment may be constructed of a variety of named and selected EnvironmentConfigs, as well as patches from the Composite, and the absence of one or more selectors may not be a problem, and may even be expected.